pkg/backup/client.go

```diff
@@ -439,44 +441,14 @@ func (bc *Client) BackupRanges(
 	t := time.NewTicker(time.Second * 5)
 	defer t.Stop()
 
-	backupTS := req.EndVersion
-	// use lastBackupTS as safePoint if exists
-	if req.StartVersion > 0 {
-		backupTS = req.StartVersion
-	}
-
-	log.Info("current backup safePoint job",
-		zap.Uint64("backupTS", backupTS))
-
-	finished := false
-	for {
-		err := UpdateServiceSafePoint(ctx, bc.mgr.GetPDClient(), bc.GetGCTTL(), backupTS)
-		if err != nil {
-			log.Error("update GC safePoint with TTL failed", zap.Error(err))
-			return nil, err
-		}
-		err = CheckGCSafePoint(ctx, bc.mgr.GetPDClient(), backupTS)
+	for err := range errCh {
 		if err != nil {
-			log.Error("check GC safePoint failed", zap.Error(err))
 			return nil, err
 		}
-		if finished {
-			// Return error (if there is any) before finishing backup.
-			return allFiles, err
-		}
-		select {
-		case err, ok := <-errCh:
-			if !ok {
-				// Before finish backup, we have to make sure
-				// the backup ts does not fall behind with GC safepoint.
-				finished = true
-			}
-			if err != nil {
-				return nil, err
-			}
-		case <-t.C:
-		}
 	}
+
+	<-allFilesCollected
+	return allFiles, nil
```

Comment:
```suggestion
	select {
	case <-allFilesCollected:
		return allFiles, nil
	case <-ctx.Done():
		return nil, errors.Trace(ctx.Err())
	}
```

maybe combine with the `errCh` too, like if `errCh` returns `nil` treat it as success.

Reply:
`errCh` is closed at this context, maybe there is no need to select on it.